halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java

```diff
@@ -26,8 +26,25 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+/**
+ * This allows fields making use of validators to be validated.
+ *
+ * This is mean to be implemented by all halconfig contents meant to be updated. It reads the Validate* anotations, and
+ * runs the validators on the fields changed.
+ *
+ * @see Validator
+ */
 public interface Updateable {
-  /* Attempts to update field to value, but will be rejected if the validation based on `context fails */
+  /**
+   * Attempts to update field to value, but will be rejected if the validation fails.
+   *
+   * @param fieldName Name of the field to be updated.
+   * @param value The value to update the named field to.
+   * @param valueType The type of value being updated (can't be inferred when value is null).
+   * @return The list of errors when validation fails (or null/[] when there are no failures).
```

Comment:
No, it's `null` throughout, but the stream processing behaves the same for `null` and `[]`. It seemed like wasted effort to validate that it's one and not the other, but I'll add the checks if you think it's valuable.